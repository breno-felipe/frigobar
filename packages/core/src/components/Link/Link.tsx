import styled from 'styled-components';

const Link = styled.a`
  text-decoration: none;

  ${({
    theme: {
      colors: { primary },
    },
  }) => `
    color: ${primary[400]};

    &:hover {
      color: ${primary[200]};
    }
  `}
`;

export default Link;
