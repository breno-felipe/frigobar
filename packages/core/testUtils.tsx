import React from 'react';
import PropTypes from 'prop-types';
import {
  render,
  fireEvent,
  waitFor,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from './src/theme';

const Providers = ({ children }: { children?: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

Providers.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, fireEvent, waitFor };
