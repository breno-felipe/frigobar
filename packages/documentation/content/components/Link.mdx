---
title: Link
menu: components
---

Link component

```jsx react-live=true
<Link href="/components/link/">Simple Link</Link>
```
