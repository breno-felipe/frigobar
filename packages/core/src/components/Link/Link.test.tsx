import React from 'react';
import { render } from '../../../testUtils';

import Link from './Link';

describe('<Link />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Link href="test.html">Simple Link </Link>);
      expect(container).toMatchSnapshot();
    });
  });
});
