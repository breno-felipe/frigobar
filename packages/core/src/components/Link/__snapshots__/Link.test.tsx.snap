// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Snapshots should match snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #269ea6;
}

.c0:hover {
  color: #80c6cb;
}

<div>
  <a
    class="c0"
    href="test.html"
  >
    Simple Link 
  </a>
</div>
`;
